common/core/web/input-processor/src/build.sh

```diff
@@ -0,0 +1,91 @@
+#! /bin/bash
+# 
+# Compile KeymanWeb's 'keyboard-processor' module, one of the components of Web's 'core' module.
+#
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../../../../resources/build/build-utils.sh"
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+# This script runs from its own folder
+cd "$(dirname "$THIS_SCRIPT")"
+
+display_usage ( ) {
+    echo "build.sh [-no-lerna | -test"
+    echo
+    echo "  -test             to compile for testing without re-fetching external dependencies"
+    echo "                    or recompiling the lm-layer module."
+    echo "  -no-lerna         skips the `lerna bootstrap` dependency check."
+    echo ""
+    echo "  If more than one target is specified, the last one will take precedence."
+    exit 1
+}
+
+# Establish default build parameters
+set_default_vars ( ) {
+    BUILD_LMLAYER=true
+    BUILD_CORE=true
+    FETCH_DEPS=true
+}
+
+set_default_vars
+
+# Parse args
+while [[ $# -gt 0 ]] ; do
+    key="$1"
+    case $key in
+        -test)
+            set_default_vars
+            BUILD_LMLAYER=false
+            FETCH_DEPS=false
+            ;;
+        -no-lerna)
+            set_default_vars
+            FETCH_DEPS=false
+            ;;
+    esac
+    shift # past argument
+done
+
+# Generates a linkable TS file; defined in resources/build-utils.sh.
+exportEnvironmentDefinitionTS
+
+if [ $FETCH_DEPS = true ]; then
+    verify_npm_setup
+fi
+
+if [ $BUILD_LMLAYER = true ]; then
+    FLAGS="-no-lerna"
+
+    # Ensure that the LMLayer compiles properly, readying the build product for comsumption by KMW.
+    cd ../../../../predictive-text/
```

Comment:
Perhaps 
```suggestion
    pushd "$(KEYMAN_ROOT)/common/predictive-text/"
```
followed by `popd` later? Makes it clearer we are returning to the input-processor folder.